heat/core/dndarray.py

```diff
@@ -730,6 +730,10 @@ def __getitem__(self, key: Union[int, Tuple[int, ...], List[int, ...]]) -> DNDar
             slices = [slice(None)] * (self.ndim - (len(kst) + len(kend)))
             key = kst + slices + kend
 
+        def __is_singular(key: any, axis: int, self_proxy: DNDarray = self_proxy) -> bool:
+            zeros = tuple([0] * (self.ndim - 1))
```

Comment:
`self_proxy.ndim` is better because the way it is, `self_proxy` can be something else (even not actually a proxy of self)